cmscontrib/loaders/lio.py

```diff
@@ -0,0 +1,322 @@
+                    logger.critical("Duplicated groups in point file")
+                    raise ValueError
+                points_per_group[group] = points
+        for group in range(len(points_per_group)):
+            if group not in points_per_group:
+                logger.critical("Missing group from point file")
+                raise ValueError
+        return points_per_group
+
+
+    def get_task(self, get_statement):
+        args = {
+            'name': self.conf['name'],
+            'title': self.conf['title'],
+        }
+        name = args['name']
+
+        logger.info(f"Loading parameters for task {name}")
+
+        if get_statement:
+            args['statements'] = {}
+            for statement in self.conf.get('statements', []):
+                path, lang = statement
+                logger.info(f"Loading statement: {statement}")
+                digest = self.file_cacher.put_file_from_path(
+                    os.path.join(self.task_dir, path),
+                    f"Statement for task {name} (lang: {lang})",
+                )
+                args['statements'][lang] = Statement(lang, digest)
+            if args['statements']:
+                args['primary_statements'] = self.conf.get('primary_statements', ["lv"])
+
+        args['submission_format'] = [f"{name}.%l"]
+
+        set_if_present(self.conf, args, 'score_precision')
+
+        score_mode = self.conf.get('score_mode', SCORE_MODE_MAX_TOKENED_LAST)
+        if score_mode in [SCORE_MODE_MAX, SCORE_MODE_MAX_SUBTASK, SCORE_MODE_MAX_TOKENED_LAST]:
+            args['score_mode'] = score_mode
+        else:
+            logger.critical("Unknown score mode provided")
+            raise ValueError
+
+        set_if_present(self.conf, args, 'max_submission_number', default=40)
+        set_if_present(self.conf, args, 'max_user_test_number', default=40)
+        set_if_present(self.conf, args, 'min_submission_interval', make_timedelta)
+        set_if_present(self.conf, args, 'min_user_test_interval', make_timedelta)
+
+        task = Task(**args)
+
+        args = {}
+        args["task"] = task
+        args["description"] = self.conf.get("version", "Default")
+        args["autojudge"] = False
+
+        args['time_limit'] = float(self.conf.get('time_limit', 2))
+        # Memory limit in MiB
+        args['memory_limit'] = self.conf.get('memory_limit', 256) * 1024**2
+
+        # Builds the parameters that depend on the task type
+        args["managers"] = {}
+
+        # By default use standard input, output
+        input_filename = self.conf.get("input_filename", "")
+        output_filename = self.conf.get("output_filename", "")
+
+        # No grader support
+        compilation_param = "alone"
+
+        if 'checker' in self.conf:
+            logger.info("Checker found, compiling")
+            checker_src = os.path.join(self.task_dir, self.conf['checker'])
+            checker_exe = os.path.join(os.path.dirname(checker_src), "checker")
```

Comment:
Here we modify the source file hierarchy, which feels a bit wrong.

If you have interest, you could try replacing this with `tempfile.NamedTemporaryFile` (the file name for GCC call can be obtained with `f.name`) and using `put_file_from_fobj` later.

Otherwise, you can leave this as is :)

Reply:
With NamedTemporaryFile there is platform dependant issues about if file can be opened the second time, I will use temporary directory instead.